optics-core/src/Optics/Internal/Indexed.hs

```diff
@@ -16,83 +15,86 @@ import Data.Profunctor.Indexed
 import Optics.Internal.Optic
 
 -- | Show useful error message when a function expects optics without indices.
-class is ~ NoIx => AcceptsEmptyIndices (f :: Symbol) (is :: IxList)
+class is ~ 'NoIx => AcceptsEmptyIndices (f :: Symbol) (is :: IxList)
 
 instance
   ( TypeError
     ('Text "‘" ':<>: 'Text f ':<>: 'Text "’ accepts only optics with no indices")
-  , (x ': xs) ~ NoIx
-  ) => AcceptsEmptyIndices f (x ': xs)
+  , xs ~ 'NoIx
```

Comment:
This doesn't look right.

Reply:
It's fine, GHC wants the equality because it's in the class definition, but it'll always show the error.